### PR TITLE
Add notification badge

### LINKED ISSUE #4365: Notification Count
Moving this over from #1785 

Here's a mockup of what it could look like:

screen shot 2014-12-08 at 2 07 46 pm

Here's the function I wrote to calculate the number, placed in the notifications ctrl js file. Since the new messages don't populate until a refresh (I think that's correct), this should work right?

```
    var newMessagesCountFunc = function() {
      var count = 0;
      if($rootScope.user.invitations.party)
        count += $rootScope.user.invitations.party.id ? 1 : 0;
      if($rootScope.user.purchased.plans)
        count += $rootScope.user.purchased.plans.mysterItems.length ? 1 : 0;
      if($rootScope.user.invitations.guilds)
        count += $rootScope.user.invitations.guilds.length;
      if($rootScope.user.flags.classSelected
         && !$rootScope.user.preferences.disableClasses
         && $rootScope.user.stats.points)
        count += $rootScope.user.stats.points > 0 ? 1 : 0;
      if($rootScope.user.newMessages) {
        for (var k in $rootScope.user.newMessages) {
          if ($rootScope.user.newMessages.hasOwnProperty(k))
            count++;
        }
      }

      return count;
    }

    $rootScope.newMessagesCount = newMessagesCountFunc();
```

The script could be better/I may have missed something. I started out by making it just a function, but then it was being called 3 times on page load and that seemed excessive. 
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/6864256-notification-count?utm_campaign=plugin&utm_content=tracker%2F68393&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F68393&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by luanmuniz**
Fixes #4365
### Changes

Add style for badge 
https://github.com/luanmuniz/habitrpg/blob/5daf8823ad820bf3b62bbd355fa359992fa0eb9f/website/client/css/menu.styl#L142-L150

Add function to MenuCtrl scope
https://github.com/luanmuniz/habitrpg/blob/5daf8823ad820bf3b62bbd355fa359992fa0eb9f/website/client/js/controllers/menuCtrl.js#L32-L64

Add badge to view
https://github.com/luanmuniz/habitrpg/blob/5daf8823ad820bf3b62bbd355fa359992fa0eb9f/website/views/shared/header/menu.jade#L180

Images:
screen shot 2016-07-02 at 21 21 56
screen shot 2016-07-02 at 21 21 50
screen shot 2016-07-02 at 21 21 44
screen shot 2016-07-02 at 21 21 38
screen shot 2016-07-02 at 21 21 31

---

UUID: 0d8208a5-9485-4a2f-9be9-009d9ed5679f

### CODE REVIEW

File: website/client/js/controllers/menuCtrl.js [hunk 1]
```diff
@@ -29,6 +29,40 @@ angular.module('habitrpg')
...
+
+        if($scope.user.newMessages) {
+          for (var k in $scope.user.newMessages) {
+            if ($scope.user.newMessages.hasOwnProperty(k)){
+              count++;
+            }
+          }
+        }
+
+        if(!count) {
```
Comment (crookedneighbor) [inline website/client/js/controllers/menuCtrl.js]: This isn't needed. You've instantiated count as 0, so returning count is returning 0.

File: website/client/js/controllers/menuCtrl.js [hunk 2]
```diff
@@ -29,6 +29,40 @@ angular.module('habitrpg')
       $scope.clearMessages = Chat.markChatSeen;
       $scope.clearCards = Chat.clearCards;
 
+      $scope.newMessagesCountFunc = function() {
```
Comment (crookedneighbor) [inline website/client/js/controllers/menuCtrl.js]: I think a better name would be `getNotificationsCount`

File: website/client/js/controllers/menuCtrl.js [hunk 3]
```diff
@@ -29,6 +29,40 @@ angular.module('habitrpg')
...
+        if($scope.user.invitations.guilds){
+          count += $scope.user.invitations.guilds.length;
+        }
+
+        if($scope.user.flags.classSelected && !$scope.user.preferences.disableClasses && $scope.user.stats.points){
+          count += $scope.user.stats.points > 0 ? 1 : 0;
+        }
+
+        if($scope.user.newMessages) {
+          for (var k in $scope.user.newMessages) {
```
Comment (crookedneighbor) [inline website/client/js/controllers/menuCtrl.js]: Could also do 

``` js
count += Object.keys($scope.user.newMessages).length;
```

**Comment by luanmuniz** [comment]
@crookedneighbor Done!

File: website/client/js/controllers/menuCtrl.js [hunk 4]
```diff
@@ -29,6 +29,32 @@ angular.module('habitrpg')
...
       $scope.clearCards = Chat.clearCards;
 
+      $scope.getNotificationsCount = function() {
+        var count = 0;
+
+        if($scope.user.invitations.party && $scope.user.invitations.party.id){
+          count++;
+        }
+
+        if($scope.user.purchased.plans && $scope.user.purchased.plans.mysterItems.length){
```
Comment (khipkin) [inline website/client/js/controllers/menuCtrl.js]: `mysteryItems`

**Comment by khipkin** [comment]
Screenshot for zero notifications?

**Comment by luanmuniz** [comment]
@khipkin The badge simple doesn't show
screen shot 2016-07-02 at 23 26 32

**Comment by luanmuniz** [comment]
Hi guys, is there anything else to update?

File: website/client/js/controllers/menuCtrl.js [hunk 5]
```diff
@@ -29,6 +29,32 @@ angular.module('habitrpg')
...
       $scope.clearCards = Chat.clearCards;
 
+      $scope.getNotificationsCount = function() {
+        var count = 0;
+
+        if($scope.user.invitations.party && $scope.user.invitations.party.id){
+          count++;
+        }
+
+        if($scope.user.purchased.plans && $scope.user.purchased.plans.mysteryItems.length){
```
Comment (crookedneighbor) [inline website/client/js/controllers/menuCtrl.js:39]: You have a typo here. It's `plan` not `plans`. I've fixed it up and squashed the commits.

**Comment by crookedneighbor** [comment]
Tier awarded! You can read more about [contributor tiers and rewards here](http://habitrpg.wikia.com/wiki/Contributor_Rewards).

Note that as your tier increases, it takes progressively more work to achieve the next one, so for the higher tiers, larger PRs or a few PRs are required. There's some information about that at Contributor Rewards, but feel free to ask here if you have questions!

**Comment by khipkin** [comment]
@luanmuniz Congrats! This feature is awesome. :tada:

**Comment by luanmuniz** [comment]
@khipkin Thank you, i'm a little proud of it. Its the first time i look into habitica code and i never worked with angular before. Its not a nice PR, a lot of typos and really simple code. But its nice to send the first one. i want to to send more. 

@crookedneighbor Sorry for the mess in this PR.

**Pull request closed without merging**